modules/ROOT/pages/understanding-enterprise-integration-patterns-using-mule.adoc

```diff
@@ -0,0 +1,92 @@
+= Understanding Enterprise Integration Patterns Using Mule
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Enterprise Integration Patterns are accepted solutions to recurring problems within a given context. The patterns provide a framework for designing, building messaging and integration systems, as well as a common language for teams to use when architecting solutions.
+
+Mule supports most of these patterns shown in the Enterprise Integration Patterns book written by Gregor Hohpe and Bobby Woolf.
+
+By implementing these patterns, Mule reduces the effort required when building integrations. You can use these well-known patters to design your solution and then simply configure and use these same patters in Mule.
+
+
+== Mapping Enterprise Integration Patterns into Mule Objects
+
+There is a list of patterns that can be mapped directly with Mule objects. Review the following list of Enterprise Integration Patterns mapped to Mule objects:
+
+=== Integration Styles
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|File Transfer |xref:connectors::file/file-connector.adoc[File Connector]
+|Shared Database |xref:connectors::db/db-connector-index.adoc[DataBase Connector]
+|Remote Procedure Invocation | Mule APIs are meant to work like this (or even doing requests to external APIs).
+|Messaging |Mule is all about Messaging.
+|===
+
+
+=== Messaging Systems
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Message Channel |Mule provides a message channel that connects the message processors in a flow.
+|Pipes and Filters |A flow implements a pipe and filter architecture.
+|Message Router |xref:about-components.adoc#flow-control-routers[Message Routers].
+|Message Translator |xref:transform-component-about.adoc[Message Transformer].
+|Message Endpoint | xref:mule-sdk::sources.adoc[Message Sources] and xref:mule-sdk::operations.adoc[Operations].
+|===
+
+=== Messaging Channels
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Point-to-Point Channel |The default channel within a flow.
+|Message Bus |Mule is a message bus.
+|Guaranteed Delivery | Using xref:reliability-patterns.adoc[Reliability Patterns].
+|===
+
+=== Message Construction
+
+[%header,cols="2*"]
+|===
+|Pattern |Mapping to a Mule Object
+|Event Message | Mule works transmitting events from different Application or Processors.
+|Request Reply | Connectors that allow request-reply wise operations, or using xref:reliability-patterns.adoc[Reliability Patterns].
```

Comment:
Mule uses connectors that facilitate request-reply operations or operations using reliability patterns.